### PR TITLE
Fix the logic for perplexity evaluation (`Not enough kv_cache capacity to run generation. Please use a larger sequence_length or a shorter prompt`)

### PR DESCRIPTION
**Pull request by dbogunowicz**

### CODE REVIEW

File: src/deepsparse/evaluation/integrations/perplexity.py [hunk 1]
```diff
@@ -165,6 +166,7 @@ def run_perplexity(
                     return_input_tokens=True,
                 )
             else:
+                print(len(pipeline.tokenizer(batch[0]).input_ids))
```
Comment (bfineran) [inline src/deepsparse/evaluation/integrations/perplexity.py:169]: debug?

File: src/deepsparse/evaluation/integrations/perplexity.py [hunk 2]
```diff
@@ -252,7 +254,15 @@ def load_perplexity_dataset(
...
         if max_sequence_length is None and pipeline is not None:
-            max_sequence_length = pipeline.sequence_length
+            # max_sequence_length for the dataset concatenation needs to be
+            # smaller than the kv_cache.capacity
+            # (pipeline.sequence_length - pipeline.prompt_sequence_length)
+            max_sequence_length = (
+                pipeline.sequence_length - pipeline.prompt_sequence_length - 1
+            )
+            # account for potential additional BOS token
+            breakpoint()
```
Comment (bfineran) [inline src/deepsparse/evaluation/integrations/perplexity.py:264]: debug?

**Review by bfineran — CHANGES_REQUESTED** [review]
looks like this may be out of date or not ready to merge @dbogunowicz?

**Comment by dbogunowicz** [comment]
Closing, due to inactivity.

**Pull request closed without merging**